### ISSUE TITLE
PHP $_SERVER array no longer includes HTTP_REFERER when using Octane on Vapor

### ISSUE THREAD

**Issue by tcampbPPU** [laravel/octane#406]
- Octane Version: `1.0.13`
- Laravel Version: `8.64.0`
- PHP Version: [`8.0`](FROM `laravelphp/vapor:php80`)
- Server & Version: RoadRunner `2.4.1`
- Database Driver & Version: RDS MySQL 8.0 

### Description:
I am testing out Octane on one of my Vapor environments. Often when doing authentication I need to use the `REFERER` from the HTTP header. When only adding Octane to my project I now get:
```
ErrorException
Undefined array key "HTTP_REFERER"
```
To verify nothing different code wise I have other environment, which is replica of env where I add Octane and I am able to do without exception:
```php
$_SERVER['HTTP_REFERER']
```

### Steps To Reproduce:

I am not able to share public code base but I can try to give small example...
Using [Socialite](https://laravel.com/docs/8.x/socialite) I have 2 routes in `api.php`

```php
// Provider in example lets say is Google
Route::get('redirect/{provider}', [AuthController::class, 'redirect']);
Route::get('callback/{provider}', [AuthController::class, 'callback']);
```
Within the `AuthController`
```php
public function redirect(string $provider): mixed
{
    return Socialite::driver($provider)
        ->stateless()
        ->redirect()
        // $_SERVER['HTTP_REFERER'] - is no longer valid option when using Octane with Vapor
        // Which is where I am getting exception
        ->withCookie(cookie('x-origin-referer', $_SERVER['HTTP_REFERER']));
}
```
TLDR;
Maybe isn't bug (maybe as general question), but with Octane do I no longer have access to `$_SERVER` info?
